### PR TITLE
Bump stack version to 7.17 in tests, recipes and samples

### PR DESCRIPTION
**Pull request by pebrc**
I also removed the dedicated 8x test job.

### CODE REVIEW

File: .ci/pipelines/e2e-tests-stack-versions-gke.Jenkinsfile [hunk 1]
```diff
@@ -183,7 +194,8 @@ pipeline {
                     "eck-711-${BUILD_NUMBER}-e2e",
                     "eck-712-${BUILD_NUMBER}-e2e",
                     "eck-713-${BUILD_NUMBER}-e2e",
-                    "eck-714-${BUILD_NUMBER}-e2e"
+                    "eck-714-${BUILD_NUMBER}-e2e",
+                    "eck-716-${BUILD_NUMBER}-e2e"
```
Comment (thbkrkr) [inline .ci/pipelines/e2e-tests-stack-versions-gke.Jenkinsfile:198]: `7.15` and `7.17`?

Comment (pebrc) [reply .ci/pipelines/e2e-tests-stack-versions-gke.Jenkinsfile:198]: Yes we did, I can add it.

**Comment by thbkrkr** [comment]
The [EMS recipe](https://github.com/elastic/cloud-on-k8s/tree/main/config/recipes/maps) still uses the 7.15.2. We missed it in #5141.

**Review by thbkrkr — APPROVED** [review]
LGTM

**Pull request merged**